pkg/sypgp/sypgp.go

```diff
@@ -612,13 +612,15 @@ func formatMROutput(mrString string, longOutput bool) (int, []byte, error) {
 	gotName := false
 	for _, l := range lines {
 		fields := strings.Split(strings.TrimSpace(l), ":")
-		if fields[0] == "info" {
+		switch fields[0] {
+
+		case "info":
 			var err error
 			numKeys, err = strconv.Atoi(fields[2])
```

Comment:
@DrDaveD this may need the `if len(fields) >=2 && fields[2] != ""`

Reply:
I see the spec saying that the "info" line is optional but I don't read it to say that the count is optional if the line is there.

Reply:
the last paragraph of section 5 states:
`Note that empty fields are allowed.  For example, a key with no
   expiration date would have the <expirationdate> field empty.  Also,
   a keyserver that does not track a particular piece of information
   may leave that field empty as well.  Colons for empty fields on the
   end of each line may be left off, if desired.`

I read it as the count is option if there are no keys returned for a given search.

Reply:
see also, the section on `info` wherein `If this optional line is not included, __or the version information is
   not supplied__, the version number is assumed to be 1`

Reply:
If there are no results then the keyserver definitely knows that so I think it is not an optional piece of information, it should return zero.

Reply:
The wording of the spec seems to indicate that any field could be empty and assuming all following fields are empty as well, colon splitting and accessing a slice at an index could cause an out of bounds panic. It doesnt make sense to me for an implementation to do this, but is appears to be allowed. I think we should do the index check here and in the `uid` case

Reply:
It doesn't make sense for some fields to ever be blank, including this one.  I didn't put checks on fingerprint, keybits, or the uid name either.  How paranoid do we have to be?  There's not a security risk.

Reply:
@DrDaveD @ikaneshiro given the spec is ambiguous, better to be safe and defensive rather than sorry -- not to argue `It doesn't make sense for some fields to ever be blank` ... making sense is not part of the spec.   ambiguous is part of the spec.  there's not much extra code involved to be safe.

Reply:
Ok, I'm out of time on this for at least the next couple of weeks and am not motivated to continue it, so feel free to replace or build on this PR as you wish.